recipes/platform.hashing/all/conanfile.py

```diff
@@ -48,22 +48,39 @@ def validate(self):
                 self.name, self.settings.compiler))
 
         elif tools.Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration("platform.Hashing/{} "
-                                            "requires C++{} with {}, "
-                                            "which is not supported "
-                                            "by {} {}.".format(
-                self.version, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler,
+            raise ConanInvalidConfiguration("{}/{} requires c++{}, "
+                                            "which is not supported by {} {}.".format(
+                self.name, self.version, self._minimum_cpp_standard, self.settings.compiler,
                 self.settings.compiler.version))
 
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, self._minimum_cpp_standard)
 
+        if self.settings.arch in ("x86", ):
+            raise ConanInvalidConfiguration("{} does not support arch={}".format(self.name, self.settings.arch))
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
Maybe move to other `package_` functions?

Reply:
`self.info` is only valid in `package_id`.

Reply:
Move the entire function. Not `self.info.header_only()`

Reply:
Ah, you meant to move the complete method.
Not to move the content.

Well, we recently started to order the methods in a recipe in the order that conan logically calls them.
conan will call `package_id` before `package_info` because `package_id` influences `self.package_folder`.